### PR TITLE
overlord,snappy: move some high level tests to overlord exercising all the managers together

### PR DESCRIPTION
**Pull request by pedronis**
This moves some high-level tests about installation and removal, and activation of the snaps on install to overlord in a form that exercises all managers while mocking out external interactions (in the style of the original snappy package tests).

Further hollow out of the snappy package, remove code about installing with activation, remove moved tests or turn them into setting LegacyInhibitHooks for now.

Needs some hack around overriding FindBootloader until we move kernel_os.go etc out and have a proper way to do that (may also want to share the fake bootloader object in a util/test package).

### CODE REVIEW

**Comment by pedronis** [comment]
retest this please

**Comment by pedronis** [comment]
retest this please

File: overlord/managers_test.go [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+	c.Assert(err, IsNil)
+	return filepath.Join(tmpdir, snapFilePath)
+
+}
+
+func makeTestSnap(c *C, snapYamlContent string) string {
+	return makeTestSnapWithFiles(c, snapYamlContent, nil)
+}
+
+func (ms *mgrsSuite) TestHappyLocalInstall(c *C) {
```
Comment (mvo5) [inline overlord/managers_test.go]: HappyLocal! :heart:

**Comment by mvo5** [comment]
Like this a lot! Thanks for brining these high level tests over. I think the hack is ok, moving kernel_os should now be easier, when I last looked the flags where a bit of a problem but we mostly sorted them.

File: overlord/managers_test.go [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+	ms.udev = testutil.MockCommand(c, "udevadm", "")
+
+	o, err := overlord.New()
+	c.Assert(err, IsNil)
+	ms.o = o
+}
+
+func (ms *mgrsSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("")
+	os.Setenv("SNAPPY_SQUASHFS_UNPACK_FOR_TESTS", "")
```
Comment (chipaca) [inline overlord/managers_test.go]: not `os.Unsetenv`?

File: snappy/snapp_snapfs_test.go [hunk 3]
```diff
@@ -292,7 +250,7 @@ func (s *SquashfsTestSuite) TestInstallKernelSnapUnpacksKernel(c *C) {
 		OfficialName: "ubuntu-kernel",
 		Revision:     snap.R(42),
 	}
-	_, err := (&Overlord{}).InstallWithSideInfo(snapPkg, si, 0, &MockProgressMeter{})
+	_, err := (&Overlord{}).InstallWithSideInfo(snapPkg, si, LegacyInhibitHooks, &MockProgressMeter{})
```
Comment (chipaca) [inline snappy/snapp_snapfs_test.go]: why these changes? Was the old test broken in that it didn't have inhibit hooks set?

Comment (pedronis) [inline snappy/snapp_snapfs_test.go]: @chipaca this branch hollows out the Overlord.Installs such that they don't do a full activation anymore (anyway it didn't make sense), and even if we need to resurrect u-d-f is passing InhbitHooks.

 As usual just trying to keep the tests going until they are ported
 while still emptying snappy progressively.

File: overlord/managers_test.go (hunk 2 above)
Comment (pedronis) [inline overlord/managers_test.go]: seems I tend to forget that that exists, or maybe it wasn't there on day 0 ? anyway will fix

**Comment by chipaca** [comment]
:+1:

File: overlord/managers_test.go (hunk 2 above)
Comment (chipaca) [inline overlord/managers_test.go]: added in 1.4
